#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()
gen.add("max_size", int_t, 0, "the max number of the points of each cluster", 25000, 0, 50000)
gen.add("min_size", int_t, 0, "the minimum number of the points of each cluster", 100, 0, 1000)
gen.add("max_area", double_t, 0, "the max area of the convex areas", 100, 0, 100)
gen.add("min_area", double_t, 0, "the minimum area of the convex areas", 0.1, 0, 1)
gen.add("angular_threshold", double_t, 0, "angular threshold of normal",
        0.04, 0, pi)
gen.add("distance_threshold", double_t, 0, "distance threshold of normal",
        0.01, 0, pi)
gen.add("max_curvature", double_t, 0, "max curvature",
        0.1, 0, pi)
gen.add("cluster_tolerance", double_t, 0, "cluster tolerance",
        0.1, 0, 500.0)
gen.add("ransac_refine_outlier_distance_threshold",
        double_t, 0, "distance threshold  for ransac refinement",
        0.1, 0.0, 0.3)
gen.add("ransac_refine_max_iterations",
        int_t, 0, "max iterations of ransac refinement",
        100, 0, 10000)
exit (gen.generate (PACKAGE, "jsk_pcl_ros",
                    "RegionGrowingMultiplePlaneSegmentation"))
